hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractPropertyMapping.java

```diff
@@ -195,14 +193,16 @@ else if ( type instanceof CollectionType ) {
 					return;
 				}
 
-				Collection commonCollection = getSuperCollection(
-						metadata,
-						thisCollection.getOwner(),
-						otherCollection.getOwner(),
-						thisCollection.getReferencedPropertyName()
-				);
-
-				newType = commonCollection.getType();
+				// When we discover incompatible types, we register "null" as property type to signal that the property is not resolvable on the parent type
+				newType = null;
+				if ( LOG.isTraceEnabled() ) {
+					LOG.tracev(
+							"Collection mapping in abstract entity type with a type variable is unsupported! Couldn't add property [{0}], existing type = [{1}], incoming type = [{2}]",
+							path,
+							existingType,
+							type
```

Comment:
am I wrong or most of the time the log does not convey the right information, like in the case of the added test?

Reply:
RIght, adapting the message to say something like "Skipping property..." would be more appropriate.

Reply:
I agree with @dreab8 here that the message should not be logged if the super does not define this attribute.  It would be confusing to me as a user to see that log message in the case Andrea brings up

Reply:
In 5.2.13 and before a "duplicate attribute" trace log is logged. I could omit the log if you like or adapt the message.

Reply:
Well @dreab8 's use case is such that it is actually not a duplicated attribute. E.g.

````
Entity(name = "BaseEntity")
@Inheritance(strategy = InheritanceType.JOINED)
public static class BaseEntity {
	@Id
	private long id;
}

@Entity(name = "EntityA")
public static class EntityA extends BaseEntity {
	@OneToMany(fetch = FetchType.LAZY)
	private Set<EntityC> attributes;
}

@Entity(name = "EntityB")
public static class EntityB extends BaseEntity {
	@OneToMany(fetch = FetchType.LAZY)
	private Set<EntityD> attributes;
}

@Entity(name = "EntityC")
public static class EntityC {
	@Id
	private long id;
}

@Entity(name = "EntityD")
public static class EntityD {
	@Id
	private long id;
}
````

Here the 2 `#attributes` attributes are not duplicate.  And I'm not so sure that the old code logged that as you say.  I could be wrong - in which case imo even the old code would have been wrong

Reply:
It's a duplicate attribute for `BaseEntity`. Note that the attributes are added to the base types as well. I agree that it's not _correct_ that it's a duplicate, but technically we are adding a duplicate. At the end of the day, this is a trace message which IMO is only for internal purposes and rarely interesting for the end user/developer.